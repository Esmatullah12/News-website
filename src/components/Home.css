
.hero {
  text-align: center;
  margin: 50px 20px 100px 20px;
}

.hero h2 {
  font-size: 3.6rem;
  font-weight: 600;
}

.hero p {
  margin: 20px auto;
  max-width: 900px;
  font-size: 1.5rem;
  font-weight: 500;
}

.hero input {
  font-size: 1rem;
  padding: 10px;
  border: 1px solid black;
  box-shadow: rgba(255, 255, 255, 0.24) 0 3px 8px;
  border-radius: 10px;
  min-width: 500px;
  background-color: rgba(255, 255, 255, 0.1);
  backdrop-filter: blur(10px);
}

.hero input:focus {
  outline: none;
}

.loading-text {
  text-align: center;
}

.news-articles {
  justify-content: center;
  border: 1px solid black;
  max-width: 90%;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 200px));
  gap: 20px;
  margin: 20px auto;
}

.article-card {
  text-decoration: none;
  color: black;
}
