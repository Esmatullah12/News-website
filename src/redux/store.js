import { configureStore } from '@reduxjs/toolkit';
import newsSlice from './news/newsSlice';

const store = configureStore({
  reducer: {
    news: newsSlice,
  },
});

export default store;
