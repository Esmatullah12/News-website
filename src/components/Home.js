import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchUsers } from '../redux/news/newsSlice';
import NewsArticle from './NewsArticle';
import './Home.css';

const Home = () => {
  const news = useSelector((state) => state.news);
  const dispatch = useDispatch();
  const [searchQuery, setSearchQuery] = useState('');
  const [filterNews, setFilterNews] = useState(news.news);
  useEffect(() => {
    dispatch(fetchUsers());
  }, [dispatch]);
  useEffect(() => {
    const filteredNews = news.news.filter((aNews) => {
      const newsTitle = aNews.newsTitle ? aNews.newsTitle.toLowerCase() : '';
      const query = searchQuery.toLowerCase();
      return newsTitle.includes(query) && aNews.newsImg;
    });
    setFilterNews(filteredNews);
  }, [news.news, searchQuery]);
  return (
    <div className="home-page">
      <div className="hero">
        <h2>News Articals</h2>
        <p>
          The Wall Street Journal recent articles from the past six months
          are sorted chronologically, offering readers instant access to the
          latest news and analyses. This efficient organization ensures
          up-to-the-minute information, enabling readers to stay well-informed
          with ease.
        </p>
        <input type="text" placeholder="Search news articles" value={searchQuery} onChange={(e) => setSearchQuery(e.target.value)} />
      </div>
      {news.loading && <div className="loading-text">Loading News...</div>}
      {!news.loading && news.error ? <div>{news.error}</div> : null}
      {!news.loading && news.news.length ? (
        <div className="news-articles">
          {filterNews.map((nNews) => (
            <Link className="article-card" key={nNews.newsId} to={`/newsDetails/${nNews.newsId}`}>
              <NewsArticle
                newsId={nNews.newsId}
                title={nNews.newsTitle}
                description={nNews.newsDesc}
                date={nNews.newsDate}
                imageUrl={nNews.newsImg}
              />
            </Link>
          ))}
        </div>
      ) : null}
    </div>
  );
};

export default Home;
