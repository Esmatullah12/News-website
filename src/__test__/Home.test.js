import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import store from '../redux/store';

describe('Test Home component', () => {
  it('it should render Home component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
