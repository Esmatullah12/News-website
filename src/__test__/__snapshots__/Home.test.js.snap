// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home component it should render Home component correctly 1`] = `
<div>
  <div
    class="home-page"
  >
    <div
      class="hero"
    >
      <h2>
        News Articals
      </h2>
      <p>
        The Wall Street Journal recent articles from the past six months are sorted chronologically.
      </p>
      <input
        placeholder="Search news articles"
        type="text"
        value=""
      />
    </div>
    <div
      class="sub-title"
    >
      The latest news appear first.
    </div>
    <div
      class="loading-text"
    >
      Loading News...
    </div>
  </div>
</div>
`;
